python/tvm/autotvm/measure/measure_methods.py

```diff
@@ -0,0 +1,344 @@
+        remote = request_remote(device_key, rpc_tracker_addr, rpc_priority, rpc_timeout)
+        remote.upload(path)
+        func = remote.load_module(file_name)
+        ctx = remote.context(str(inp.target), 0)
+        time_f = func.time_evaluator(
+            func.entry_name, ctx, number=number, repeat=repeat)
+        return time_f, ctx, args
+
+    ret = _measure_generic(_fbuild, input_pack, check_correctness)
+    return ret
+
+
+def measure_local(input_pack,
+                  number,
+                  repeat=1,
+                  check_correctness=False,
+                  build_option=None):
+    """Measure the time cost on a local machine.
+
+    Parameters
+    ----------
+    input_pack : list of MeasureInput
+        The inputs we need to evaluate
+    number : int
+        Number of times to get the running measurement
+    repeat : int, optional
+        How many times we want to repeat the measurement.
+    check_correctness: bool, optional
+        Whether check correctness after measurement.
+    build_option: dict, optional
+        build options for tvm.build_config
+
+    Returns
+    -------
+    res_pack : Array of MeasureResult
+        The list of execution results of measurement.
+    """
+
+    def _fbuild(inp):
+        """ Local build function """
+        with inp.target:
+            s, args = inp.task.instantiate(inp.config)
+            if not inp.config.valid():
+                raise InstantiationError(inp.config.errors)
+            code_hash = getattr(s, 'code_hash', None)
+            if inp.config.code_hash != code_hash:
+                raise HashMismatchError('got {0:s}, expected {1:s}'
+                                        .format(str(inp.config.code_hash), str(code_hash)))
+
+            opts = build_option or {}
+            ctx = context(str(inp.target), 0)
+            if "cuda" in inp.target.keys:
+                # Add cuda verify pass to filter out invalid configs in advance.
+                # This can accelerate the tuning process
+                opts["add_lower_pass"] = \
+                    [(2, cuda_verify_pass(ctx.max_shared_memory_per_block,
+                                          ctx.max_threads_per_block))]
+            with build_config(**opts):
+                func = build(s, args, target_host=inp.task.target_host)
+
+        time_f = func.time_evaluator(
+            func.entry_name, ctx, number=number, repeat=repeat)
+        return time_f, ctx, args
+
+    ret = _measure_generic(_fbuild, input_pack, check_correctness)
+    return ret
+
+
+def cuda_verify_pass(**kwargs):
+    """Verify the invalidity of a cuda kernel
+    This pass will check shared memory size and number of thread per block.
+    """
+    def verify_pass(stmt):
+        valid = ir_pass.VerifyGPUCode(stmt, kwargs)
+        if not valid:
+            raise InstantiationError("invalid cuda kernel")
+        return stmt
+    return verify_pass
+
+
+_cuda_target_arch = None
```

Comment:
avoid use of global-keyword, can use a singleton scope e.g. AutoTVMEnv (Like build config), to store necessary info